specs/fast-cli.md

```diff
@@ -0,0 +1,164 @@
+- [Abstract](#abstract)
+- [Use Cases](#use-cases)
+- [Requirements](#requirements)
+    - [Configuration](#configuration)
+        - [Project Type](#project-type)
+            - [Application](#application)
+            - [Component Library](#component-library)
+            - [Non-Specific Project Type Options](#non-specific-project-type-options)
+    - [Command Line](#command-line)
+        - [Initialization](#initialization)
+        - [Adding A Component](#adding-a-component)
+            - [Adding A Foundation Based Component](#adding-a-foundation-based-component)
+    - [Structure](#structure)
+        - [Create FAST Project Package](#create-fast-project-package)
+        - [FAST CLI Package](#fast-cli-package)
+    - [Dependencies](#dependencies)
+- [Implementation](#implementation)
+- [Documentation](#documentation)
+    - [System Requirements](#system-requirements)
+    - [Use Of The CLI](#use-of-the-cli)
+    - [How To Create Templates](#how-to-create-templates)
+- [Maintenance](#maintenance)
+    - [Testing](#testing)
+- [Roadmap](#roadmap)
+    - [MVP](#mvp)
+
+## Abstract
+
+This specification describes a CLI used for the creation and running of FAST based applications and component libraries called `@microsoft/fast-cli`, as well as a package leveraging the CLI and allowing `npm` to initialize it called `@microsoft/create-fast-project`. The CLI is intended to accelerate user workflows when using FAST, and the initialization package is intended to assist in a quick setup for projects such as applications and component libraries.
+
+## Use Cases
+
```

Comment:
I'm not sure if this falls into the "prototype" bucket, but a common scenario in the WC world is that I just want to create a single element as a package. This could also be the same scenario where we have a user that just wants to try out FAST locally with a super simple, basic project. Would these fall under prototype or would they be a separate concern?

Reply:
i think a "prototype" use case is subjective to the project type, you could be prototyping an app, or you could be prototyping a component library, both cases are covered below. Even if it's a single component, I believe if it is intended to be used as a package you'd choose component library.